File: cpp/robot-name/robot_name.cpp
#include "robot_name.h"

using namespace std;

namespace robot_name
{
    robot::robot()
    {
        _name = generate_name();
    }

    string robot::name() const
    {
        return _name;
    }

    void robot::reset()
    {
        _name = generate_name();
    }

    string robot::generate_name() const
    {
        random_device generator;

        uniform_int_distribution<char> letters('A', 'Z');
        uniform_int_distribution<char> digits('0', '9');

        return string{
            letters(generator),
            letters(generator),
            digits(generator),
            digits(generator),
            digits(generator),
        };
    }
} // namespace robot_name

File: python/robot-name/robot_name.py
from typing import List
import random


class Robot:
    def __init__(self):
        self._name: str = self.__generate_name()

    @property
    def name(self) -> str:
        return self._name

    def reset(self) -> None:
        self._name = self.__generate_name()

    def __generate_name(self) -> str:
        random.seed()
        letters: List[str] = random.sample("ABCDEFGHIJKLMNOPQRSTUVWXYZ", 2)
        digits: List[str] = random.sample("0123456789", 3)

        return "".join(letters + digits)
